src/integrationTest/java/io/aiven/kafka/connect/jdbc/oracle/VerifyInsertIT.java

```diff
@@ -0,0 +1,216 @@
+                final String recordValue = "value-" + i;
+                final ProducerRecord<String, GenericRecord> msg = createRecord(i, partition, recordName, recordValue);
+                sendFutures.add(producer.send(msg));
+            }
+        }
+        producer.flush();
+        for (final Future<RecordMetadata> sendFuture : sendFutures) {
+            sendFuture.get();
+        }
+    }
+
+    @AfterEach
+    public void afterEach() throws SQLException {
+        executeSqlStatement(DROP_TABLE);
+    }
+
+    @Test
+    public void testSinkConnector() throws Exception {
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        executeSqlStatement(CREATE_TABLE);
+
+        // Start the sink connector
+        connectRunner.createConnector(basicSinkConnectorConfig());
+
+        // Send test data to Kafka topic
+        sendTestData(1);
+
+        await().atMost(Duration.ofSeconds(6)).pollInterval(Duration.ofSeconds(2))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(1);
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).column("ID")
+                            .value().isEqualTo("0");
+                });
+    }
+
+    @Test
+    public void testSinkWithPKModeRecordKeyConnector() throws Exception {
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        executeSqlStatement(CREATE_TABLE_WITH_PK);
+
+        // Start the sink connector
+        connectRunner.createConnector(sinkConnectorConfigWith_PKModeRecordKey());
+
+        // Send test data to Kafka topic
+        sendTestData(1);
+
+        await().atMost(Duration.ofSeconds(6)).pollInterval(Duration.ofSeconds(2))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(1);
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).column("ID")
+                            .value().isEqualTo("0");
+                });
+    }
+
+    @Test
+    public void testSinkConnectorInDeleteMode() throws Exception {
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        executeSqlStatement(CREATE_TABLE);
+
+        // Start the sink connector
+        connectRunner.createConnector(sinkConnectorConfigWith_PKModeRecordKey_DeleteEnabled());
+
+        // Send test data to Kafka topic
+        sendTestData(1);
+
+        await().atMost(Duration.ofSeconds(6)).pollInterval(Duration.ofSeconds(2))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(1);
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).column("ID")
+                            .value().isEqualTo("0");
+                });
+    }
+
+    @Test
+    public void testMultiInsertMode() throws Exception {
```

Comment:
This name is a little misleading:
```suggestion
    public void testMultiInsertInDeleteMode() throws Exception {
```